### PR TITLE
Converted .rst files to MyST .md files

### LINKED ISSUE #155026: Convert to markdown: mobile_optimizer.rst, model_zoo.rst, module_tracker.rst, monitor.rst, mps_environment_variables.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [mobile_optimizer.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/mobile_optimizer.rst)
- [model_zoo.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/model_zoo.rst)
- [module_tracker.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/module_tracker.rst)
- [monitor.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/monitor.rst)
- [mps_environment_variables.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/mps_environment_variables.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

### PR DESCRIPTION
**Pull request by FORTFANOP**
Fixes #155026

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Review by zxiiro — DISMISSED** [review]

**Comment by svekars** [comment]
Please use `git mv filename.rst filename.md` as per the instructions in the issue to convert instead of deleting the .rst file and creating an .md file.

File: docs/source/model_zoo.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# torch.utils.model_zoo
+
+Moved to `torch.hub`.
+
```
Comment (svekars) [inline docs/source/model_zoo.md:4]: Probably need to add:
```suggestion

```{currentmodule} torch.utils.model_zoo
```
```

File: docs/source/mobile_optimizer.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (svekars) [inline docs/source/mobile_optimizer.md:1]: I think this file might have been improperly converted. If I run from your branch/fork:
```
git log --follow --oneline -- docs/source/mobile_optimizer.md
```
I only get one item in the git log:
```
71c7477c84f convert mobile_optimizer.rst to mobile_optimizer.md
```
But if I run from main for the .rst file I get:
```
0d193a0adf0 Add ExecuTorch warning to mobile_optimizer (#134697)
da43584bef3 [Reland] Clean Up MobileOptimizerType Rewrite Flags Public API and Documentation (#92081)
4f91b8e0ee8 Fix typo under docs directory  (#91871)
3aeb7127b44 Revert "Clean Up MobileOptimizerType Rewrite Flags Public API and Documentation (#91600)"
370df963e06 Clean Up MobileOptimizerType Rewrite Flags Public API and Documentation (#91600)
df1cc0ef473 [Vulkan] Add Vulkan Rewrite to Transfer Inputs and Outputs to Vulkan and CPU Backends Respectively (#87432)
b3387139b42 Mod lists to neutral+descriptive terms in caffe2/docs (#49803)
79b8328aaf5 optimize_for_mobile: bring packed params to root module (#42740)
04320a47d76 Add optimizer_for_mobile doc into python api root doc (#41211)
```
We want to preserve history for this file which I think we achieve (more or less) with `git mv` or any other way you want to propose. All other files I checked in this PR seem to have preserved their git log.

Comment (FORTFANOP) [reply docs/source/mobile_optimizer.md:1]: That's weird, I used the same method for all files, ie, git mv
On my local repo too, running the command you mentioned doesn't work for the .md files but they do for the .rst files. Not sure why this is so, and looks like the same issue is happening with others as well

Comment (FORTFANOP) [reply docs/source/mobile_optimizer.md:1]: Kindly forgive if its trivial, I'm new to open source

Comment (zxiiro) [reply docs/source/mobile_optimizer.md:1]: Its due to how Git works. If too many lines in a file is changed Git won't find similarity between 2 files so doesn't show that they were renamed. Git doesn't actually store the history of changes in its database but rather the state of the entire workspace. I tried to explain it here hopefully it helps explain but this is expected.

https://github.com/pytorch/pytorch/pull/155120#issuecomment-2943883096
